src/vm/compiler.c

```diff
@@ -2073,7 +2073,22 @@ static void ifStatement(Compiler *compiler) {
 
     patchJump(compiler, endJump);
 }
-
+static void matchStatement(Compiler *compiler) {
+    consume(compiler, TOKEN_LEFT_PAREN, "Expect '(' after 'match'.");
+    expression(compiler);
+    consume(compiler, TOKEN_RIGHT_PAREN, "Expect ')' after expression.");
+    consume(compiler, TOKEN_LEFT_BRACE, "Expect '{' before match body.");
+    consume(compiler, TOKEN_CASE, "Expect atleast one 'case' block.");
+      do{
+      expression(compiler);
+      int compareJump = emitJump(compiler, OP_COMPARE_JUMP);
+      consume(compiler, TOKEN_COLON, "Expect ':' after expression.");
+      statement(compiler);
+      patchJump(compiler,compareJump);
+      }while(match(compiler, TOKEN_CASE)); 
```

Comment:
Small formatting suggestions.

```suggestion
    do {
        expression(compiler);
        int compareJump = emitJump(compiler, OP_COMPARE_JUMP);
        consume(compiler, TOKEN_COLON, "Expect ':' after expression.");
        statement(compiler);
        patchJump(compiler, compareJump);
    } while(match(compiler, TOKEN_CASE)); 
```